File: erlang/raindrops/src/raindrops.erl
-module(raindrops).

-export([convert/1]).


convert(Number,_ , Done) when ((Number rem 3) == 0) and ((Done rem 3) /= 0) ->
    convert(Number div 3,"Pling", Done * 3);
convert(Number,Res, Done) when ((Number rem 5) == 0) and ((Done rem 5) /= 0) ->
    convert(Number div 5,string:concat(Res,"Plang"), Done * 5);
convert(Number,Res, Done) when ((Number rem 7) == 0) and ((Done rem 7) /= 0) ->
    convert(Number div 7,string:concat(Res,"Plong"),Done * 7);
convert(Number,"", _) -> integer_to_list(Number);
convert(_,Res, _) -> Res.
convert(Number) ->
    convert(Number,"", 1).

File: go/raindrops/raindrops.go
package raindrops

import "strconv"

// Convert an integer into a raindrops
func Convert(n int) string {
	rains := map[int]string{3: "Pling", 5: "Plang", 7: "Plong"}
	res := ""
	for k, v := range rains {
		if n%k == 0 {
			res += v
		}
	}

	if res == "" {
		return strconv.Itoa(n)
	} else {
		return res
	}
}
